api/src/main/java/io/opentelemetry/trace/attributes/BooleanAttribute.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.attributes;
+
+import io.opentelemetry.trace.Span;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/** Defines the behavior for a span attribute with boolean values. */
+@Immutable
+public class BooleanAttribute extends AbstractAttribute<Boolean> {
+
+  /**
+   * Constructs an attribute object.
+   *
+   * @param attributeKey the attribute name/key
+   */
+  public BooleanAttribute(String attributeKey) {
+    super(attributeKey);
+  }
+
+  @Override
+  public void set(Span span, @Nullable Boolean value) {
+    if (value != null) {
+      span.setAttribute(key(), value);
+    } else {
+      span.setAttribute(key(), (String) null);
```

Comment:
This case looks really awkward. Why would we convert a boolean to a string? Isn't setting a null string equivalent to not calling `setAttribute` as opposed to clearing the attribute anyway?

Reply:
Changed